src/AbstractMigrator.php

```diff
@@ -0,0 +1,395 @@
+     */
+    protected function _getMigrationSqlQuery($currVersion, $increment)
+    {
+        $currVersion = $this->_normalizeInt($currVersion);
+        $fileNumber  = ($increment >= 0)
+            ? $currVersion + 1
+            : $currVersion;
+
+        $file = $this->getMigrationSql($fileNumber, $increment);
+
+        return file_exists($file) && is_readable($file)
+            ? file_get_contents($file)
+            : null;
+    }
+
+    /**
+     * Retrieves the path to an SQL file.
+     *
+     * Overridden to use the new directory customization.
+     *
+     * @since [*next-version*]
+     *
+     * @param int|string $version   The version.
+     * @param int        $increment The increment.
+     *
+     * @return string|null The path to the SQL file or null if no matching file was found.
+     */
+    public function getMigrationSql($version, $increment)
+    {
+        $version = $this->_normalizeInt($version);
+        $files   = $this->_getMigrationFiles($version, $increment);
+        $count   = count($files);
+
+        if ($count > 1) {
+            throw $this->_createCouldNotMigrateException(
+                $this->__('Found multiple migration files with the same version number'),
+                null,
+                null,
+                $version
+            );
+        }
+
+        return $count
+            ? reset($files)
+            : null;
+    }
+
+    /**
+     * Retrieves the migration files that match the given version and increment.
+     *
+     * @since [*next-version*]
+     *
+     * @param int $version   The migration version.
+     * @param int $increment The increment, either 1 or -1.
+     *
+     * @return string[] The matched migration file paths.
+     */
+    protected function _getMigrationFiles($version, $increment)
+    {
+        $results   = [];
+        $direction = ($increment < 0)
+            ? static::MIGRATION_DIRECTION_DOWN
+            : static::MIGRATION_DIRECTION_UP;
+
+        $patterns = $this->_getMigrationFilePatterns($direction);
+
+        foreach ($patterns as $_dir => $_pattern) {
+            $_regex  = sprintf($_pattern, $version);
+            $_files  = $this->_getMatchingFiles($_dir, $_regex);
+            $results = array_merge($results, $_files);
+        }
+
+        return $results;
+    }
+
+    /**
+     * Retrieves the path of all files in a directory that match a given regex pattern.
+     *
+     * @since [*next-version*]
+     *
+     * @param string $directory
+     * @param        $regex
+     *
+     * @return array
+     */
+    protected function _getMatchingFiles($directory, $regex)
+    {
+        $directory = $this->_normalizeString($directory);
+        $regex     = $this->_normalizeString($regex);
+
+        if (!is_dir($directory)) {
+            return [];
+        }
+
+        $files  = [];
+        $handle = opendir($directory);
+
+        if ($handle) {
+            while (($_file = readdir($handle)) !== false) {
```

Comment:
Maybe, this can be simplified using [`RegexIterator`] and [`DirectoryIterator`].

[`RegexIterator`]: http://php.net/manual/en/class.regexiterator.php
[`DirectoryIterator`]: http://php.net/manual/en/class.directoryiterator.php

Reply:
I don't think I know how to accomplish this. But I agree that it would probably be a better solution that the current one. 

Also, I think a third iterator will be needed to modify matching entries. The current code transforms the file names into absolute paths.

Inti provajd gaidinse?

Reply:
Take a look at [this example].

[this example]: https://stackoverflow.com/questions/20752387/get-specific-files-in-a-recursive-directory